src/utils/dataFormatter.ts

```diff
@@ -1,25 +1,35 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
 import { DynamoDBRecord } from 'aws-lambda';
+import { DynamoDB } from 'aws-sdk';
 import { TestActivity } from './testActivity';
 
-export const dataFormatter = (record: DynamoDBRecord): TestActivity => {
-  const data = record.dynamodb as TestActivity;
+export const formatDynamoData = (record: DynamoDBRecord): TestActivity[] => {
+  const data = DynamoDB.Converter.unmarshall(record.dynamodb.NewImage);
+  console.log(data);
 
-  const activityEvent: TestActivity = {
-    noOfAxles: data.noOfAxles,
-    testTypeStartTimestamp: data.testTypeStartTimestamp,
-    testTypeEndTimestamp: data.testTypeEndTimestamp,
-    testStationType: data.testStationType,
-    testCode: data.testCode,
-    vin: data.vin,
-    vrm: data.vrm,
-    testStationPNumber: data.testStationPNumber,
-    testResult: data.testResult,
-    certificateNumber: data.certificateNumber,
-    testTypeName: data.testTypeName,
-    vehicleType: data.vehicleType,
-    testerName: data.testerName,
-    testerStaffId: data.testerStaffId,
-    testResultId: data.testResultId,
-  };
-  return activityEvent;
+  const testActivities: Array<TestActivity> = [];
+
+  data.testTypes.forEach((testType) => {
```

Comment:
Nice! Because you have 1 : 1 mapping between input and output you can probably use a .map here and you don't need the temporary array or the pushes

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
Sounds good, just about to commit the changes